### PR TITLE
add config & mastosearch, update mastoarch & README

### PR DESCRIPTION
**Pull request by IzzySoft**
* add a new script to ease searches (`mastosearch`)
* add a config file to define defaults for `mastoarch` and `mastosearch`
* update contrib's README accordingly

Gimmick: with `mastosearch`, using the `viewer` parameter (and having the `markdown` package installed) results can be shown in a browser that accepts input from STDIN – e.g. lynx(see example). Not yet tested: other browsers. There are some [examples for Firefox](https://unix.stackexchange.com/q/24931/209279), but that might be overdoing it :wink:

### CODE REVIEW

**Comment by kensanata** [comment]
Soon, it will spin up a local HTTP server, start the default browser, and have some JavaScript to filter the results… and once you have that, perhaps you can write a viewer from scratch! :laughing:

**Comment by IzzySoft** [comment]
Sure, Python has that option integrated. So one day, just specify `viewer="mastodon-backup --server"` :yum:

I'm just testing out the things in this PR (found no issues during the past 2 days, which is why I now opened it). But if you feel it's stable enough, be welcome to remove the "Draft" and merge :smiley:

**Comment by IzzySoft** [comment]
OK, crontab is now established (hence the last commit above, that made me apply another small update – though I'm not sure if it was really *required* as it seams that `read` is simply ignored when there's no TTY, but better have it clean & clear). Tried different variants of it to see the output, now set it for a nightly backup with output redirected to `/dev/null`. If there are no complaints until tomorrow, I'll remove the "Draft" state here and declare it ready.

**Comment by IzzySoft** [comment]
Couldn't resist, sorry :see_no_evil: follower-search for now just returns the matched JSON records, no formatting. Had no idea for something simple turning it into "nice HTML", hints welcome. (note: that ugly redirect `$(</dev/stdin)` could also be replaced by `$(cat)` if preferred).

Cron job worked fine btw.

File: contrib/README.md [hunk 1]
```diff
@@ -37,15 +37,34 @@ It will work in the directory you called it from. So if you wish to use it, I'd
 recommend you either put it into your `$PATH` or create yourself an alias for it.
 
 Once you've set it up and have it running, you could even have a Cron job
-taken care for regular runs. Here I'd recommend the following command line:
+taken care for regular runs. With the config set up properly, the command
+for that job could look like this:
 
 ```bash
-cd /path/to/archive && mastoarch -a Me@MyInstance 2>/dev/null
+source /home/johndoe && mastoarch archive
```
Comment (kensanata) [inline contrib/README.md:44]: I don't understand `source /home/johndoe` – doesn't that attempt to read `/home/johndoe` as shell script?

File: contrib/mastoarch [hunk 2]
```diff
@@ -8,20 +8,33 @@
...
+archive_followers=0     # whether to archive followers as well
 autogit=0               # whether to automatically check-in new data to your local git repo
 
 # !!! ================================================================== !!!
 # !!! Do not change below this line unless you know what you're doing :) !!!
 # !!! ================================================================== !!!
 
+# -----------------------------------------------------------------------------
+# Load config if exists
+[[ -f "${HOME}/.config/mastodon-backup/config" ]] && source "${HOME}/.config/mastodon-backup/config"
```
Comment (kensanata) [inline contrib/mastoarch:20]: I know that the path on GitHub still says mastodon-backup but everywhere else the name has been changed to mastodon-archive.

File: contrib/README.md [hunk 3]
```diff
@@ -37,15 +37,34 @@ It will work in the directory you called it from. So if you wish to use it, I'd
...
+## `mastosearch` (by Izzy)
+For your impromptu searches. Again a wrapper around the main script intended
+to be easy to use. With your defaults set up in the configuration, a search
+is as easy as `mastosearch <searchterm>`. Again, the full syntax is shown when
+called without parameters.
+
+## `config.sample` (by Izzy)
+Lets you define defaults to use with `mastoarch` and `mastosearch` to make
+calls to them as simple as possible. On your Linux machine, copy it to
+`${HOME}/.config/mastodon-backup/config` and adjust it to fit your needs.
```
Comment (kensanata) [inline contrib/README.md:65]: mastodon-archive, as discussed below

Comment (kensanata) [reply contrib/README.md:65]: Another option would be to check if the config file exists and if it does not, querying the user for values.

File: contrib/README.md (hunk 1 above)
Comment (IzzySoft) [reply contrib/README.md:44]: Oh… By replacing my login a part got lost, that should be `/home/johndoe/.profile`. Adjusted (commit coming).

File: contrib/mastoarch (hunk 2 above)
Comment (IzzySoft) [reply contrib/mastoarch:20]: So shall I change all paths accordingly? `${HOME}/.config/mastodon-archive` as config dir, for consistency? No big deal at this point (as I'm most likely the only one using it currently).

File: contrib/README.md (hunk 3 above)
Comment (IzzySoft) [reply contrib/README.md:65]: The latter is done. Default values are defined in the script, then the config is loaded if it exists, then command-line parameters are processed. The only interactive question is whether to commit to git, and that only if `autobackup=1` (not with `0` or `2`). Adding a hint at this place.

**Comment by IzzySoft** [comment]
I assume the DEB/RPM packages should be renamed to `mastodon-archive` then as well?

File: contrib/README.md (hunk 3 above)
Comment (IzzySoft) [reply contrib/README.md]: The former is done as well. If you are OK with how I handle defaults (avoiding interactively prompting the users), this conversation could then be resolved.

**Comment by kensanata** [comment]
Thanks!

**Comment by kensanata** [comment]
Yeah, renaming the packages would be better. I'm a bit unsure about the redirect. Any git repo with a remote pointing here will be in trouble, no? Or does git protocol handle redirects, too? I've never tried it.

**Comment by IzzySoft** [comment]
> Yeah, renaming the packages would be better.

Done, will add the new packages to their issue ASAP.

> I'm a bit unsure about the redirect. Any git repo with a remote pointing here will be in trouble, no?

No. From my experience as one of the F-Droid maintainers: I sometimes note a redirect is in place only months later and "by accident" (I have a script checking for new apps to add to my own repo, which filters out those already present at F-Droid or in mine by comparing the source code URLs. After a move, they no longer match and the app shows up as "hey, I'm new" with my script. All the while, updates at F-Droid performed their `git pull` fine on the old URL).

> Or does git protocol handle redirects, too?

Going by the above it looks like it does. To make sure, you could create some dummy repo, clone it, push some dummy commits, then move it and try pulling from the original address. Afterwards simply drop it. Give it some stupid name you think you'll never need in the near future (like "test123"), just to make sure not to block something – I've no idea how Github handles names :see_no_evil:

**Pull request merged**